components/org.wso2.carbon.identity.application.authenticator.google/src/main/java/org/wso2/carbon/identity/application/authenticator/google/GoogleOAuth2Authenticator.java

```diff
@@ -51,13 +69,191 @@ public class GoogleOAuth2Authenticator extends OpenIDConnectAuthenticator {
      * Initiate tokenEndpoint
      */
     private void initTokenEndpoint() {
+
         this.tokenEndpoint = getAuthenticatorConfig().getParameterMap().get(GoogleOAuth2AuthenticationConstant
                 .GOOGLE_TOKEN_ENDPOINT);
         if (StringUtils.isBlank(this.tokenEndpoint)) {
             this.tokenEndpoint = IdentityApplicationConstants.GOOGLE_TOKEN_URL;
         }
     }
 
+    @Override
+    public boolean canHandle(HttpServletRequest request) {
+        /*
+          Validity of the
+          1. CSRF cookies
+          2. JWT token
+          decide the ability of handling the authentication request in Google One Tap flow.
+         */
+        if (isOneTapEnabled(request)) {
+            boolean validCookies = validateCSRFCookies(request);
+            if (validCookies) {
+                return validateJWTFromGOT(request);
+            }
+        }
+        return super.canHandle(request);
+    }
+
+    /**
+     * This function validates the JWT token by its content using Google libraries.
+     *
+     * @param request Authentication request with JWT token.
+     * @return Validity of the returned JWT token returned via Google One Tap.
+     */
+    private boolean validateJWTFromGOT(HttpServletRequest request) {
+
+        ExternalIdPConfig externalIdPConfig = null;
+        try {
+            String authenticatorName = request.getParameter(IDP);
+            String tenantDomain = String.valueOf(request.getAttribute(TENANT_DOMAIN));// Returns "null" for null
```

Comment:
Can assign this to an `Optional`. Do we need to handle null?

Reply:
Fixed 76bde03e8a2edc30459231f28a0cf90d9bbee240
These lines were removed in latest modifications